nextcord/client.py

```diff
@@ -2102,33 +2117,170 @@ def get_application_command(self, command_id: int) -> Optional[BaseApplicationCo
+
+        .. versionchanged:: 2.4
+            Added the ``search_locales`` keyword argument.
+
+        .. versionchanged:: 2.4
+            Subcommands/Subcommand groups can now be retrieved with this method.
+
+        .. versionchanged:: 2.4
+            Changed the signature from ``(qualified_name, type, guild_id, *, search_locales)`` to ``(*, type, qualified_name, guild_id, search_locales)``.
+            All parameters are now keyword-only.
+
         Parameters
         ----------
-        name: :class:`str`
-            Name of the application command. Capital sensitive.
-        cmd_type: Union[:class:`int`, :class:`ApplicationCommandType`]
+        type: Union[:class:`int`, :class:`ApplicationCommandType`]
             Type of application command.
+        qualified_name: :class:`str`
+            Full name of the application command. Case sensitive.
+            Subcommands must be separated by a space, E.g, ``parent group subcommand``.
         guild_id: Optional[:class:`int`]
             Guild ID of the signature. If set to ``None``, it will attempt to get the global signature.
+        search_locales: :class:`bool`
+            Whether to also search through the command's name locales. Defaults to ``False``.
 
         Returns
         -------
-        command: Optional[:class:`BaseApplicationCommand`]
-            Application Command with the given signature. If no command with that signature is
-            found, returns ``None`` instead.
+        command: Optional[:class:`BaseApplicationCommand`, :class:`SlashApplicationSubcommand`]
+            Application Command with the given signature. If no command with that signature is found, ``None`` is returned instead.
         """
-        if isinstance(cmd_type, ApplicationCommandType):
-            actual_type = cmd_type.value
-        else:
-            actual_type = cmd_type
-
         return self._connection.get_application_command_from_signature(
-            name=name, cmd_type=actual_type, guild_id=guild_id
+            type=type.value if isinstance(type, ApplicationCommandType) else type,
+            qualified_name=qualified_name,
+            guild_id=guild_id,
+            search_locales=search_locales,
+        )
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        *,
+        type: Union[int, ApplicationCommandType],
+        qualified_name: str,
+        guild_id: Optional[int],
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        qualified_name: str,
+        type: Union[int, ApplicationCommandType],
+        guild_id: Optional[int],
+        *,
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @overload
+    def get_application_command_from_signature(
```

Comment:
Why is this overload here? The original `get_application_command_from_signature` never supported this function signature, so I don't see a point in adding it if positional arguments are deprecated for this method.

Reply:
Unsure about that since the overloads are making this a non-breaking change and the overload you mention is the new signature but minus the kwargs only. Also, the code supports it 🤷🏼

Reply:
Well why do we need a new posargs signature if the old posargs signature is supposed to be deprecated?

Reply:
It isn't a breaking change to remove type annotations, as people using type checkers should be able to know their signature is deprecated.

Reply:
Hmm yeah that's true